### PR TITLE
Rebuild: Make mobile sidebar work again

### PR DESCRIPTION
**Pull request by Munter**
- Fixed broken urls in section headings
- Fixed active marker of the section and avoided marking a sections index page as active when a sub page was active
- Hooked close button up
- Made drag-to-open work
- Made outside click to close menu work

### CODE REVIEW

File: src/components/SidebarMobile/SidebarMobile.jsx [hunk 1]
```diff
@@ -40,15 +39,15 @@ export default class SidebarMobile extends React.Component {
 
   componentDidMount() {
     if (typeof window !== 'undefined') {
-      // window.addEventListener('click', this._handleBodyClick);
-      window.addEventListener('touchstart', this._handleBodyClick);
+      window.addEventListener('mousedown', this._handleBodyClick);
+      // window.addEventListener('touchstart', this._handleBodyClick);
```
Comment (montogeek) [inline src/components/SidebarMobile/SidebarMobile.jsx]: Not necessary?

Comment (Munter) [reply src/components/SidebarMobile/SidebarMobile.jsx]: Doesn't seem to be. But I've only been able to test with emulated mobile in chrome dev tools

Comment (montogeek) [reply src/components/SidebarMobile/SidebarMobile.jsx]: Uhmm, we need to test in real devices

**Comment by Munter** [comment]
Can I get some help getting this tested? It's the last feature blocking the merge of the rebuild branch into master, at least if that checklist is still correct.

**Comment by montogeek** [comment]
No, there is still more work to do before merging master, like implementing `js-with-link-details` which I am working on.

**Comment by montogeek** [comment]
Thanks for fixing this @Munter.
Found 2 problems:
1. 
image
`z-index` problem, happens in Safari and Chrome.
2. https://gfycat.com/OptimalImpoliteGrouper
When you open sidebar and scroll, it closes automatically, it only happens in Safari and not Chrome.

**Comment by EugeneHlushko** [comment]
@montogeek hi mate, it works on my phone and chrome simulator now 👍

File: src/components/SidebarMobile/SidebarMobile.jsx [hunk 2]
```diff
@@ -37,18 +39,9 @@ export default class SidebarMobile extends React.Component {
...
-  }
-
-  componentWillUnmount() {
-    if (typeof window !== 'undefined') {
-      window.removeEventListener('mousedown', this._handleBodyClick);
-      // window.removeEventListener('touchstart', this._handleBodyClick);
-    }
+  _toggleBodyListener(add) {
+    let actionName = add ? 'addEventListener' : 'removeEventListener';
+    window[actionName]('mousedown', this._handleBodyClick);
```
Comment (montogeek) [inline src/components/SidebarMobile/SidebarMobile.jsx]: Did it build?

**Comment by montogeek** [comment]
Will pull it and test later

File: src/components/SidebarMobile/SidebarMobile.jsx (hunk 2 above)
Comment (EugeneHlushko) [reply src/components/SidebarMobile/SidebarMobile.jsx]: yes

Comment (EugeneHlushko) [reply src/components/SidebarMobile/SidebarMobile.jsx]: if you mean `npm run build`

Comment (montogeek) [reply src/components/SidebarMobile/SidebarMobile.jsx]: oh yeah, i was confused with something, nvm

**Comment by EugeneHlushko** [comment]
Please dont take too long, maybe we can finish off rebuild ;]

**Comment by EugeneHlushko** [comment]
Did you have a chance to try it out yet? @montogeek 
@byzyk you can also test this one Bohdan?

**Comment by montogeek** [comment]
Thanks @Munter and @EugeneHlushko

**Comment by byzyk** [comment]
@EugeneHlushko gonna test it in a couple of days. Have you guys considered Android devices as well?

**Pull request merged**